js/shared.js

```diff
@@ -1156,7 +1156,31 @@ export const validPhoneNumberFormat =
   }
 
 /**
- * Detect if current device is mobile
+ * Check if current device is a mobile device (smartphone, tablet, or others with touch screen)
  * @returns {boolean}
  */
-  export const isMobile = /Mobi/.test(navigator.userAgent);
+function checkDeviceMobile() {
+  let isMobile = false;
+
+  if ('maxTouchPoints' in navigator) {
+    isMobile = navigator.maxTouchPoints > 0;
+  } else if ('msMaxTouchPoints' in navigator) {
```

Comment:
should "else if" be on next line? on first glance almost looks like it's nested within "if" block.. just a styling preference from my point of view, so if you prefer this that's totally cool